app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -44,38 +49,54 @@ def callback_for(provider)
       cta_variant: cta_variant,
     )
 
-    if persisted_and_valid?
+    if user_persisted_and_valid?
+      # TODO: [thepracticaldev/oss] remove this
       # delete legacy session based cookie once the user has logged in again
       request.cookie_jar.delete(LEGACY_COOKIE_NAME) if request.cookies[LEGACY_COOKIE_NAME]
```

Comment:
We switched to the new cookie in October 2019 in #4004 

I think we can remove this code, let me know if there are any objections

Reply:
DO IT!